lib/vfscore/main.c

```diff
@@ -2061,37 +2061,44 @@ UK_SYSCALL_R_DEFINE(int, access, const char*, pathname, int, mode)
 
 UK_SYSCALL_R_DEFINE(int, faccessat, int, dirfd, const char*, pathname, int, mode, int, flags)
 {
-	if (flags & AT_SYMLINK_NOFOLLOW) {
-		UK_CRASH("UNIMPLEMENTED: faccessat() with AT_SYMLINK_NOFOLLOW");
-	}
+	char p[PATH_MAX];
+	struct vfscore_file *fp;
+	int error;
 
-	if (pathname[0] == '/' || dirfd == AT_FDCWD) {
-		return uk_syscall_r_access((long) pathname, (long) mode);
-	}
+	if (pathname[0] == '/' ||
+	    dirfd == AT_FDCWD) {	/* Full path inside pathname */
+		strlcpy(p, pathname, PATH_MAX);
```

Comment:
One minor comment: your change causes `faccessat` to always copy the path to a stack temporary even if it's a full (absolute) path that you could pass on to other functions as-is.
A more economical solution would be something like
```c
char tmp[PATH_MAX];
const char *p;

if (pathname[0] == '/' || dirfd == AT_FDCWD) {
    p = pathname;
} else {
    /* strcat the path together into tmp */
    p = tmp;
}
```
then use `p` as usual.
Or, if you're feeling particularly clever:
```c
const int fullpath = pathname[0] == '/' || dirfd == AT_FDCWD;
char tmp[fullpath ? PATH_MAX : 1];

if (fullpath) {
    p = pathname;
} else {
    /* build path in tmp */
    p = tmp;
}
```
in order to save stack space when not needing `tmp`.

Reply:
First Issue: You are right. I changed the code exactly as you said.
Second Part: I moved the declaration of `tmp` inside the `else` branch. The compiler should be clever with it. I hope it is ok.

Reply:
IIRC it is undefined behavior (or at the very _least_ very non-standard-C) to leak the address of a local variable outside of its scope -- that's explicitly why I didn't propose declaring `tmp` in the else clause, because you need it in the top-level function block.

If the 2nd option looks too hacky declaring `tmp` with a runtime size, you can leave it always at `PATH_MAX` and rely on the compiler being clever enough to not actually do a `sub %rsp` unless `tmp` is used. That at least will be well-defined behavior.

Reply:
I understand. I moved the declaration back to the top-level, with `PATH_MAX` size.

Reply:
Changes look good, thanks! Could you please squash them into the first commit and force-push the PR branch? Then I can give approval and we can merge.

As a policy we try to keep the commit history clean and reflecting self-contained changes, rather than tracking every adjustment during review -- so force-pushes are the common way to address comments.